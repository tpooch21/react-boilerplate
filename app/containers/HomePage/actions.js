import { LOAD_STRINGS, LOAD_STRINGS_SUCCESS } from './constants';

export const loadStrings = () => ({
  type: LOAD_STRINGS
});

export const loadStringsSuccess = () => ({
  type: LOAD_STRINGS_SUCCESS,
  strings,
});
