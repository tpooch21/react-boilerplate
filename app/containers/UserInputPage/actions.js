import { ENTER_STRING } from './constants';

export const enterString = input => ({
  type: ENTER_STRING,
  currentString: input,
});
