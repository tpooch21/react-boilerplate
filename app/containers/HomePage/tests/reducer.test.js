import homePageReducer from '../reducer';
import { loadStrings, loadStringsSuccess, loadStringsError } from '../actions';

describe('homePageReducer', () => {
  it('should return the initial state', () => {
    expect(homePageReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the loadStrings action', () => {
    expect(homePageReducer({}, loadStrings())).toMatchSnapshot();
  });

  it('handles the loadStringsSuccess action', () => {
    const strings = ['welcome'];
    expect(homePageReducer({}, loadStringsSuccess(strings))).toMatchSnapshot();
  });

  it('handles the loadStringsError action', () => {
    expect(homePageReducer({}, loadStringsError())).toMatchSnapshot();
  });
});
