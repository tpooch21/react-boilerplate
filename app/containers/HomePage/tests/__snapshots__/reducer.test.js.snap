// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homePageReducer handles the loadStrings action 1`] = `
Object {
  "error": false,
  "loading": true,
}
`;

exports[`homePageReducer handles the loadStringsError action 1`] = `
Object {
  "error": true,
  "loading": false,
}
`;

exports[`homePageReducer handles the loadStringsSuccess action 1`] = `
Object {
  "error": false,
  "loading": false,
  "strings": Array [
    "welcome",
  ],
}
`;

exports[`homePageReducer should return the initial state 1`] = `
Object {
  "error": false,
  "loading": false,
  "strings": Array [],
}
`;
